File: frontend/src/app/modules/models/post.ts
import {User} from "./user";

export class Post {
  private _id: number;
  private _photo: any;
  private _text: string;
  private _date: string;
  private _place: string;
  private _user_id: number;
  private _user: User;


  get user(): User {
    return this._user;
  }

  set user(value: User) {
    this._user = value;
  }


  constructor(id: number, photo: any, text: string, date: string,
              place: string, user_id: number, user: User) {
    this._id = id;
    this._photo = photo;
    this._text = text;
    this._date = date;
    this._place = place;
    this._user_id = user_id;
    this._user = user;
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get photo(): any {
    return this._photo;
  }

  set photo(value: any) {
    this._photo = value;
  }

  get text(): string {
    return this._text;
  }

  set text(value: string) {
    this._text = value;
  }

  get date(): string {
    return this._date;
  }

  set date(value: string) {
    this._date = value;
  }

  get place(): string {
    return this._place;
  }

  set place(value: string) {
    this._place = value;
  }

  get user_id(): number {
    return this._user_id;
  }

  set user_id(value: number) {
    this._user_id = value;
  }
}

File: backend/src/main/java/com/netcracker/backend/entity/Post.java
package com.netcracker.backend.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;

import javax.persistence.*;
import java.util.List;


@Entity
public class Post implements Comparable<Post>{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name ="id")
    private long id;

    @Lob
    @Column(name = "photo")
    private String photo;
    public String getPhoto() {
        return photo;
    }
    public void setPhoto(String photo) {
        this.photo = photo;
    }

    @Basic
    @Column(name = "text")
    private String text;
    @Basic
    @Column(name = "date")
    private String date;
    @Basic
    @Column(name = "place")
    private String place;

    @JsonBackReference(value="user-post")
    @ManyToOne
    @JoinColumn(name = "userId")
    private User user;
    public User getUser() {
        return user;
    }
    public void setUser(User user) {
        this.user = user;
    }



    @JsonManagedReference(value="comment-post")
    @OneToMany(mappedBy = "post")
    private List<Comment> comments;

    @JsonManagedReference(value="like-post")
    @OneToMany(mappedBy = "post")
    private List<Likes> likes;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }


    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public String getPlace() {
        return place;
    }

    public void setPlace(String place) {
        this.place = place;
    }

    public Post(){};
    public Post(String date, long id, String photo, String place, String text, User user) {
        this.date = date;
        this.id = id;
        this.photo = photo;
        this.place = place;
        this.text = text;
        this.user = user;
    }


    @Override
    public int compareTo(Post o) {
        return (this.getDate().compareTo(o.getDate()))*(-1);
    }

}
